articles/language/statements.md

```diff
@@ -149,45 +150,25 @@ Q# distinguishes between mutable and immutable symbols.
 In general, the use of immutable symbols is encouraged because it
 allows the compiler to perform more optimizations.
 
-For arrays, mutability or immutability applies both to the array as a whole
-and to array elements.
-That is, the elements of an immutable array may not be changed.
-
-The contents of a tuple instance or of a user-defined type based on a tuple
-may not be changed regardless of the mutability of the symbol bound to the
-instance.
-If a mutable symbol is bound to a tuple, the symbol may be bound
-to a new tuple, but the existing tuple may not be modified in place.
+The left-hand-side of the binding consists of a symbol tuple, and the right hand side of an expression.
 
-The arguments to an operation or a function are treated as immutable.
-This also applies to elements of an array passed to an operation or function.
-This means that Q# does not support "out" variables.
+Since all Q# types are value types - with the qubits taking a somewhat special role - 
+formally a "copy" is created when a value is bound to a symbol, or when a symbol is rebound. 
+That is to say, the behavior of Q# is the same as if a copy were created on assignment. 
+This in particular also includes arrays. Of course in practice only the relevant pieces are actually recreated as needed. 
 
 ### Tuple Deconstruction
 
-If the right-hand side of the binding is a tuple or a user-defined type,
-then an extended syntax may be used to deconstruct the tuple:
-
-```qsharp
-let (i, f) = (5, 0.1);
-```
-
-This statement will bind `5` to `i` and `0.1` to `f`.
-
-Deconstructing binds may also be used for more complex tuples, such as:
-
-```qsharp
-mutable (a, (b, c)) = (1, (2, 3));
-set (x, y) = (1, (2, 3));
-```
-
-In these examples, `a` and `x` both get bound to `1`,
-`b` gets bound to `2`, `c` to `3`, and `y` to `(2, 3)`.
-
-Tuple deconstruction can also be used when the right-hand side of the `=`
-is a tuple-valued expression:
+If the right-hand side of the binding is a tuple,
+then that tuple may be deconstructed upon assignment.
+Such deconstructions may involve nested tuples, and any full or partial deconstruction is valid as long as the shape of the tuple on the right hand side is compatible with the shape of the symbol tuple.
+Tuple deconstruction can in particular also be used when the right-hand side of the `=` is a tuple-valued expression.
 
 ```qsharp
+let (i, f) = (5, 0.1); // i is bound to 5 and f to 0.1
+mutable (a, (_, b)) = (1, (2, 3)); // a is bound to 1, b is bound to 3
+mutable (x, y) = ((1, 2), [3, 4]); // x is bound to (1,2), y is bound to [3,4]
+set (x, _, y) = ((5, 6), 7, [8]);  // x is rebound to (5,6), y is rebound to [8]
 let (r1, r2) = MeasureTwice(q1, PauliX, q2, PauliY);
 ```
 
```

Comment:
Perhaps this is mentioned elsewhere (if so, my apologies – please disregard), but it might be helpful to mention that the assignment to the iteration variable for a `for` loop can also be a tuple deconstruction.

Reply:
yes, that is meantioned elsewhere